server/controllers/users.js

```diff
@@ -194,16 +204,26 @@ export const exchangeLoginToken = async (req, res, next) => {
 
   const twoFactorAuthenticationEnabled = parseToBoolean(config.twoFactorAuthentication.enabled);
   if (twoFactorAuthenticationEnabled && (await twoFactorAuthLib.userHasTwoFactorAuthEnabled(req.remoteUser))) {
+    const supported2FAMethods = [
+      TwoFactorMethod.RECOVERY_CODE,
+      ...(await twoFactorAuthLib.twoFactorMethodsSupportedByUser(req.remoteUser)),
+    ];
```

Comment:
The reason is that a recovery code option is always available on the API if the user has 2FA enabled, but we dont want to present it on the frontend for every action since using the code resets all methods for the user.
Here its the login route, so we want to present it, I think an alternative would be to always return recovery code as an option and have this decision about presenting it on the frontend based on context.